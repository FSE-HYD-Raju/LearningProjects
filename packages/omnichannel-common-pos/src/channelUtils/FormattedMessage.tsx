"use strict";

import * as React from "react";
import { omit } from "lodash";
import { FormattedMessage as IntlFormattedMessage } from "react-intl";

import IntlContainer from "./IntlContainer";

interface FormattedMessageProps extends IntlFormattedMessage.Props {
	wrapperType?: "wrapper";
	className?: string; // Optional class name. Applied to a <span> with resulting message if present
}

/**
 * A wrapper on react-intl FormattedMessage that bundles it into IntlContainer (to ease usage).
 *
 * Please note that if className prop is passed - custom render function is discarded.
 *
 * Actually there is no need in className when custom render function is passed as one can provide any html
 * directly into that function.
 */
const FormattedMessage: React.FC<FormattedMessageProps> = (props) => {
	const messageProps = omit(props, "wrapperType", "className");

	if (props.className) {
		return (
			<IntlContainer>
				<IntlFormattedMessage {...messageProps} >
					{(txt) => (
						<span className={props.className}>
							{txt}
						</span>
					)}
				</IntlFormattedMessage>
			</IntlContainer>
		);
	}

	return (
		<IntlContainer>
			<IntlFormattedMessage {...messageProps} />
		</IntlContainer>
	);
};
FormattedMessage.displayName = "FormattedMessageWrapper";

FormattedMessage.defaultProps = {
	wrapperType: "wrapper"
};

export default FormattedMessage;
export type FormattedMessageDescriptor = IntlFormattedMessage.MessageDescriptor;
export {
	FormattedMessageProps,
};
