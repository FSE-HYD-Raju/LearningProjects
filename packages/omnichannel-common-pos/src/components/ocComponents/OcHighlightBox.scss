@import '../../styles/_variables.scss';

.OcHighlightBox {
        background: $body-bg;
        border: 0px solid #4287B6;

    &-top {
        border-width: 3px 0 0 0;
        >div {
            border: 1px solid #E1E1E1;
            border-top: transparent;
        }
    }

    &-right {
        border-width: 0 3px 0 0;
        >div {
            border: 1px solid #E1E1E1;
            border-right: transparent;
        }
    }

    &-bottom {
        border-width: 0 0 3px 0;
        >div {
            border: 1px solid #E1E1E1;
            border-bottom: transparent;
        }
    }

    &-left {
        border-width: 0 0 0 3px;
        >div {
            border: 1px solid #E1E1E1;
            border-left: transparent;
        }
    }
}
