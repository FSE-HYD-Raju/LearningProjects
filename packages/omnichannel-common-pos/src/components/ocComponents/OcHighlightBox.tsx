import * as R from "react";
import cssns from "../../utils/cssnsConfig";
import * as classnames from "classnames";
const React = cssns("OcHighlightBox").React as typeof R;

interface OcHighlightBoxProps {
	children?: React.ReactNode;
	className?: string;
	position?: "top" | "right" | "bottom" | "left";
	style?: React.CSSProperties;
}

const OcHighlightBox: React.FC<OcHighlightBoxProps> = props => {
	const { children, className, position, style } = props;
	const classNames = classnames("this", {
		top: position === "top",
		right: position === "right",
		bottom: position === "bottom",
		left: position === "left" || !position,
		[`${className}`]: !!className
	});

	return (
		<div className={classNames} style={{ ...style }}>
			<div>{children}</div>
		</div>
	);
};

export default OcHighlightBox;
export { OcHighlightBoxProps };
