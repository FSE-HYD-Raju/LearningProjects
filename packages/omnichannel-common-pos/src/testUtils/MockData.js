import _ from "lodash";
import R from "ramda";

const firstDateThisWeek = () => {
	const date = new Date();
	while (date.getDay() !== 0) {
		date.setDate(date.getDate() - 1);
	}
	return date;
};

const people = [
	{
		firstName: "Kathrin",
		lastName: "K",
		gender: "female",
		language: "swe",
		countryOfBirth: "SE",
		birthDay: firstDateThisWeek(),
		emails: [{ email: "kathrin.k@qvantel.com" }],
		email: "kathrin.k@qvantel.com",
		mobileNumbers: [{ number: "2398457" }],
		mobileNumber: "2398457",
		fixedLineNumbers: [{ number: "2398457" }],
		fixedLineNumber: "2398457",
		password: "qwerasdf",
		passwordConfirmation: "qwerasdf"
	},
	{
		firstName: "John",
		lastName: "Smith",
		gender: "male",
		language: "fin",
		birthDay: firstDateThisWeek(),
		countryOfBirth: "United Kingdom",
		emails: [{ email: "john.smith@qvantel.com" }],
		email: "john.smith@qvantel.com",
		mobileNumbers: [{ number: "3498573" }],
		mobileNumber: "3498573",
		fixedLineNumbers: [{ number: "52497584" }],
		fixedLineNumber: "52497584"
	}
];

const basketItems = [
	{
		id: "7879c247-9e98-40ee-bb48-544eb366494a",
		type: "basketItems",
		attributes: {
			product: {
				name: "Movie Steaming Service"
			},
			totalPrices: [
				{
					type: "ONE_TIME",
					taxFreeAmount: 1.9,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: 14.9,
					taxRate: 0,
					currency: "EUR"
				}
			],
			childBasketItems: [],
			totalUpfrontPrice: 1.9,
			totalUpfrontPriceWithChildren: 1.9
		}
	},
	{
		id: "c0065ffc-77f4-4e44-b15c-fbd2f78d04ce",
		type: "basketItems",
		attributes: {
			product: {
				name: "XDSL"
			},
			totalPrices: [
				{
					type: "ONE_TIME",
					taxFreeAmount: 100.6,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: 30,
					taxRate: 0,
					currency: "EUR"
				}
			],
			childBasketItems: []
		}
	}
];

const deliveryItem = {
	id: "c0065ffc-77f4-4e44-b15c-fbd2f78d04ce",
	type: "basketItems",
	inputtedCharacteristics: {
		CH_Delivery_Contact_Info: "5c5dddf6-72be-4d0a-9711-6027f247af28"
	},
	product: {
		name: "Home delivery",
		id: "PO_Home_Delivery"
	}
};

export default class MockData {
	static getPerson(searchCriteria) {
		if (searchCriteria) {
			return _.find(people, searchCriteria);
		} else {
			return people[0];
		}
	}

	static getPeople() {
		return people;
	}

	static getPostalAddress() {
		return {
			street: "1 Test St",
			coAddress: "Best Ltd.",
			postalCode: "90210",
			city: "Test City",
			country: "FI"
		};
	}

	static getUser() {
		return {
			"id": "bf588945-ce95-490a-a774-8ef91661a146",
			"type": "persons",
			"attributes": {
				"lastName": "Testeri",
				"addresses": null,
				"gender": "male",
				"fixedLineNumbers": [],
				"formattedName": "Teppo Testeri",
				"roles": null,
				"billingAccountId": null,
				"language": null,
				"emails": [
					{
						"role": "PRIMARY",
						"id": "2ab79d7b-862d-49df-8afc-e6104a914919",
						"email": "tepsteri@tepinposti.fi"
					}
				],
				"honorificSuffix": null,
				"countryOfBirth": "FI",
				"personRelationships": [],
				"birthDay": null,
				"placeOfBirth": null,
				"characteristics": null,
				"mobileNumbers": [
					{
						"role": "PRIMARY",
						"id": "a843f46d-f729-40dd-a7ce-240a64bc008e",
						"number": "1234576890"
					}
				],
				"honorificPrefix": null,
				"avatar": "/img/icon-user-male.svg",
				"privacySettings": {
					"privacy1": false,
					"privacy2": false
				},
				"postalAddresses": [
					{
						"role": "PRIMARY",
						"id": "5e30221d-ad19-4043-8088-f594f5fdf77f",
						"street": "Piippukatu 11 A 1",
						"coAddress": null,
						"postalCode": "40100",
						"city": "Roma",
						"country": "FI",
						"county": null,
						"description": null,
						"postOfficeBox": null,
						"stateOrProvince": null,
						"addressRegisterId": null,
						"apartment": null,
						"building": null
					},
					{
						"role": "DELIVERY",
						"id": "5c5dddf6-72be-4d0a-9711-6027f247af28",
						"street": "Piippukatu 11 A 1",
						"coAddress": "Piippukatu 11 A 1",
						"postalCode": "40100",
						"city": "Roma",
						"country": "FI",
						"county": null,
						"description": null,
						"postOfficeBox": null,
						"stateOrProvince": "",
						"addressRegisterId": null,
						"apartment": null,
						"building": null
					}
				],
				"identifications": [],
				"firstName": "Teppo",
				"customerAccountId": null,
				"nationality": null,
				"additionalName": null,
				"maritalStatus": null
			}
		}
	}

	static getBasket(withItems: boolean, withDeliveryItems: boolean) {
		const basket = {
			activeBasket: {
				attributes: {
					totalPrices: [],
					totalUpfrontPrice: 0
				}
			}
		};

		if (withItems) {
			basket.basketItems = basketItems;

			const totalPriceOneTime = R.reduce(
				R.add,
				0,
				basketItems.map(bi => {
					return bi.attributes.totalPrices[0].taxFreeAmount;
				})
			);
			const totalPriceMonthly = R.reduce(
				R.add,
				0,
				basketItems.map(bi => {
					return bi.attributes.totalPrices[0].taxFreeAmount;
				})
			);

			basket.activeBasket.attributes.totalPrices = [
				{
					type: "ONE_TIME",
					taxFreeAmount: totalPriceOneTime,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: totalPriceMonthly,
					taxRate: 0,
					currency: "EUR"
				}
			];

			if (withDeliveryItems) {
				basket.basketItems[0].attributes.childBasketItems.push(
					deliveryItem
				);
			}
		}

		return basket;
	}
}
