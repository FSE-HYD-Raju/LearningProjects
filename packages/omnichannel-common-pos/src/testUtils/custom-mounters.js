import React from "react";
import { IntlProvider, intlShape } from "react-intl";
import { mount, shallow } from "enzyme";
import PropTypes from "prop-types";
import R from "ramda";
import TestUtils from "./TestUtils";

const MAX_DIVE_DEEP_LEVEL = 16;
const intlProvider = new IntlProvider({ locale: "en" }, {});
const { intl } = intlProvider.getChildContext();

const baseContext = {
	flux: {
		persist: () => {}
	},
	store: TestUtils.mockReduxStore(),
	intl,
	location: {},
	only: {},
	skip: {}
};

const nodeWithIntlProp = (node, mockedRouterOptions) => {
	if (mockedRouterOptions) {
		mockedRouterOptions.props().history.push = jest.fn();
	}
	return React.cloneElement(node, {
		intl,
		...(mockedRouterOptions ? mockedRouterOptions.props() : {})
	});
};

function getOptionsForShallowMount(options, debug, mockedRouterContextOptions) {
	const mountOptions = R.merge(options ? options : {}, {
		disableLifecycleMethods: !(
			options && options.disableLifecycleMethods === false
		),
		context: R.mergeAll([
			debug
				? {
						...baseContext
					}
				: baseContext,
			R.omit("intl", options && options.context ? options.context : {}),
			mockedRouterContextOptions.context
		]),
		childContextTypes: R.merge(
			{
				intl: intlShape,
				store: PropTypes.object
			},
			mockedRouterContextOptions.childContextTypes
		)
	});
	TestUtils.mockReduxStoreToFlux(mountOptions.context);
	return mountOptions;
}
function deepDive(wrapper) {
	let diveAttemptsCount = 0;
	while (
		wrapper.hostNodes().length === 0 &&
		diveAttemptsCount < MAX_DIVE_DEEP_LEVEL
	) {
		wrapper = wrapper.dive();
		diveAttemptsCount += 1;
	}
	return wrapper;
}
export function shallowWithContextAndRouterProps(node, options, debug) {
	return shallowWithContext(node, options, debug, true);
}
export function shallowWithContextAndLifecycle(node, options, debug) {
	return shallowWithContext(
		node,
		R.merge(options || {}, { disableLifecycleMethods: false }),
		debug,
		true
	);
}
export function deepShallowWithContextAndLifecycle(node, options, debug) {
	return deepDive(shallowWithContextAndLifecycle(node, options, debug));
}
export function deepShallowWithContext(node, options, debug) {
	return deepDive(shallowWithContext(node, options, debug, true));
}
export function shallowWithContext(node, options, debug, injectRouterProps) {
	const mockedRouterOptions = TestUtils.getMockedRouterOptions();
	const mockedRouterContextOptions = mockedRouterOptions.get();
	return shallow(
		nodeWithIntlProp(node, injectRouterProps ? mockedRouterOptions : null),
		getOptionsForShallowMount(options, debug, mockedRouterContextOptions)
	);
}

export function mountWithContextAndRouterProps(node, options, debug) {
	return mountWithContext(node, options, debug, true);
}
export function mountWithContext(node, options, debug, injectRouterProps) {
	const mockedRouterOptions = TestUtils.getMockedRouterOptions();
	const mockedRouterContextOptions = mockedRouterOptions.get();
	const mountOptions = R.merge(options ? options : {}, {
		context: R.mergeAll([
			debug
				? {
						...baseContext
					}
				: baseContext,
			R.omit("intl", options && options.context ? options.context : {}),
			mockedRouterContextOptions.context
		]),
		childContextTypes: R.merge(
			{
				flux: PropTypes.object,
				store: PropTypes.object,
				intl: intlShape,
				location: PropTypes.object,
				only: PropTypes.object,
				skip: PropTypes.object
			},
			mockedRouterContextOptions.childContextTypes
		)
	});
	TestUtils.mockReduxStoreToFlux(mountOptions.context);
	return mount(
		nodeWithIntlProp(node, injectRouterProps ? mockedRouterOptions : null),
		mountOptions
	);
}

const attachmentElementId = "__OC_COMPONENT_TEST_ATTACHMENT_ELEMENT__";

export function attachWithContext(node, options) {
	const attachmentElement = document.createElement("div");
	attachmentElement.setAttribute("id", attachmentElementId);
	document.body.appendChild(attachmentElement);

	const wrapper = mountWithContext(
		node,
		R.merge(options, { attachTo: attachmentElement })
	);
	const doppelWrapper = hijackDetacher(wrapper);

	return doppelWrapper;
}

function hijackDetacher(wrapper) {
	const doppel = wrapper;
	doppel.__detach = wrapper.detach;

	doppel.detach = () => {
		removeAttachmentElements();
		doppel.__detach.apply(wrapper);
	};

	return doppel;
}

function removeAttachmentElements() {
	document.body.querySelectorAll("#" + attachmentElementId).forEach(elem => {
		document.body.removeChild(elem);
	});
}

export function detach(wrapper) {
	removeAttachmentElements();
	wrapper.detach();
}
