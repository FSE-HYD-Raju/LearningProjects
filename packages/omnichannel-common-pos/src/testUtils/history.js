import isClient from "../utils/isClient";
import { createBrowserHistory, createMemoryHistory } from "history";
import { useRouterHistory } from "react-router";

const history = isClient
	? useRouterHistory(createBrowserHistory)({ queryKey: false })
	: createMemoryHistory();

export default history;
