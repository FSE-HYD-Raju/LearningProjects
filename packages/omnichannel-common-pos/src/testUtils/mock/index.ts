import { default as productOffering } from "./ProductOfferingMock";
import { ProductMock as product } from "./ProductMock";
import { AgreementMock as agreement } from "./AgreementMock";
import { ChargingBalancesMock as chargingBalances } from "./ChargingBalancesMock";
import { default as postalAddress } from "./PostalAddressMock";
import { default as identification } from "./IdentificationMock";
import { default as user } from "./UserMock";
import { default as order } from "./OrderMock";
import { default as basket } from "./BasketMock";
export { productOffering, postalAddress, identification, user, order, basket, product, agreement, chargingBalances };
