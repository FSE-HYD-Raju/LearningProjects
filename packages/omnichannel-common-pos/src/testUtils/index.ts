import ReactWidgetsTestUtils from "./ReactWidgetsTestUtils";
import * as MockDataMaker from "./mock";
const {
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach
} = require("./custom-mounters");
const TestUtils = require("./TestUtils");
const MockData = require("./MockData");
const SimpleDataMock = require("./mock/simpleDataMock");

export {
	MockData,
	ReactWidgetsTestUtils,
	TestUtils,
	SimpleDataMock,
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach,
	MockDataMaker
};

export * from "./MockRouterProps";

export default TestUtils;
