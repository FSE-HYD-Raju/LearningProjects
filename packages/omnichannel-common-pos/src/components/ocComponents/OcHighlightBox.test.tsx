import * as React from "react";
import { mountWithContext, shallowWithContext } from "../../testUtils";
import OcHighlightBox from "./OcHighlightBox";
import FormattedMessage from "../../channelUtils/FormattedMessage";

describe("OcHighlightBox", () => {
	it("succeeds at shallow mount without props", () => {
		// @ts-ignore
		const wrapper = shallowWithContext(<OcHighlightBox />);
		expect(wrapper).toMatchSnapshot();
	});

	it("succeeds at deep mount without props", () => {
		// @ts-ignore
		const wrapper = mountWithContext(<OcHighlightBox />);
		expect(wrapper).toMatchSnapshot();
	});

	const highlightBorder = (border: "top" | "right" | "bottom" | "left") =>
		mountWithContext(<OcHighlightBox position={border} />);

	it("should highlight the top border", () => {
		expect(
			highlightBorder("top").find(".OcHighlightBox-top").length
		).toEqual(1);
	});

	it("should highlight the right border", () => {
		expect(
			highlightBorder("right").find(".OcHighlightBox-right").length
		).toEqual(1);
	});

	it("should highlight the bottom border", () => {
		expect(
			highlightBorder("bottom").find(".OcHighlightBox-bottom").length
		).toEqual(1);
	});

	it("should highlight the left border", () => {
		expect(
			highlightBorder("left").find(".OcHighlightBox-left").length
		).toEqual(1);
	});

	it("should apply classes from props", () => {
		const props = {
			className:
				"OcHighlightBox-are-belong-to-us OcHighlightBox-foo-bar OcHighlightBox-all-your-base"
		};

		const wrapper = mountWithContext(<OcHighlightBox {...props} />);

		props.className.split(" ").forEach(className => {
			expect(wrapper.find("." + className).length).toBeGreaterThan(0);
		});
	});

	it("should apply styles from props", () => {
		const props = {
			style: {
				borderWidth: "3px 0 0 0",
				borderColor: "#E1E1E1" /* same as rgb(225, 225, 225) */
			}
		};

		const wrapper = mountWithContext(<OcHighlightBox {...props} />);

		const element = wrapper.find(".OcHighlightBox");
		const styles = element.prop("style");
		expect(styles).toBeDefined();
		// @ts-ignore
		expect(styles.borderWidth).toMatch(/^3px 0 0/);
		// @ts-ignore
		expect(styles.borderColor).toBeDefined();
		// @ts-ignore
		expect(styles.borderColor.toLowerCase()).toEqual("#e1e1e1");
	});

	it("should render with children", () => {
		const props = {
			children: (
				<React.Fragment>
					<FormattedMessage
						key="1"
						id="child-1"
						description="OcHighlightBox child 1"
						defaultMessage="Child 1"
					/>
					<FormattedMessage
						key="2"
						id="child-2"
						description="OcHighlightBox child 2"
						defaultMessage="Child 2"
					/>
				</React.Fragment>
			)
		};

		const wrapper = mountWithContext(<OcHighlightBox {...props} />);

		const children = wrapper
			.childAt(0)
			.children()
			.children();
		expect(children.length).toEqual(2);
		expect(children.at(0).text()).toEqual("Child 1");
		expect(children.at(1).text()).toEqual("Child 2");
	});
});
