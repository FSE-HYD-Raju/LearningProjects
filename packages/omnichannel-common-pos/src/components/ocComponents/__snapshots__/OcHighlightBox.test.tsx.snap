// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OcHighlightBox succeeds at deep mount without props 1`] = `
<OcHighlightBox
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <div
    className="OcHighlightBox OcHighlightBox-left"
    style={Object {}}
  >
    <div />
  </div>
</OcHighlightBox>
`;

exports[`OcHighlightBox succeeds at shallow mount without props 1`] = `
<div
  className="OcHighlightBox OcHighlightBox-left"
  style={Object {}}
>
  <div />
</div>
`;
